### ISSUE TITLE
Use floating point values in `new Color()`

### ISSUE THREAD

**Issue by fatcerberus** [spheredev/neosphere#69]
Currently `new Color()` is simply an alias for `CreateColor()` and is hardcoded to use 8-bit integer components (0-255).  Change the `Color` constructor to use floats (0-1) instead; if `CreateColor()` is retained, ensure that API remains compatible with Sphere 1.x.

Also add `.r`, `.g`, `.b` and `.a` properties to the `Color` object which are expressed in floating point, as opposed to `.red`, `.green`, `.blue` and `.alpha` which go from 0-255.

This would be a breaking change, and so is a minisphere 4.0 enhancement at the earliest.

**Comment by fatcerberus** [comment]
It might also be worth it to add read-only properties for common colors: `Color.Red`, `Color.Black`, etc.

**Comment by fatcerberus** [comment]
Predefined colors have been implemented and the feature is already in master.  I went above the call of duty with this one and implemented the entire X11 color set, so you get such colors as `Color.DodgerBlue`, `Color.Chartreuse`, `Color.Goldenrod`, and so on along with the standard red, blue, magenta, white, etc.

As adding static properties to the `Color` constructor doesn't break compatibility, that feature will be available in minisphere 3.3.  Floating point color support will still need to be held off until 4.0, though.
